programs/manifest/src/state/market.rs

```diff
@@ -0,0 +1,1077 @@
+                get_helper::<RBNode<RestingOrder>>(dynamic, current_order_index).get_value();
+            trace!("PRE MATCH rem:{remaining_base_atoms} now:{now_slot} order:{other_order:?}");
+
+            // Remove the other order if expired.
+            if other_order.is_expired(now_slot) {
+                let other_is_bid: bool = other_order.get_is_bid();
+
+                let amount_atoms_to_return: u64 = if other_is_bid {
+                    other_order
+                        .get_price()
+                        .checked_quote_for_base(other_order.get_num_base_atoms(), false)?
+                        .as_u64()
+                } else {
+                    other_order.get_num_base_atoms().as_u64()
+                };
+
+                // Global order balances are accounted for on the global accounts, not on the market.
+                if other_order.is_global() {
+                    let global_trade_accounts_opt: &Option<GlobalTradeAccounts> = if other_is_bid {
+                        &global_trade_accounts_opts[1]
+                    } else {
+                        &global_trade_accounts_opts[0]
+                    };
+                    try_to_remove_from_global(&global_trade_accounts_opt, amount_atoms_to_return)?
+                } else {
+                    update_balance_by_trader_index(
+                        dynamic,
+                        other_order.get_trader_index(),
+                        !other_is_bid,
+                        true,
+                        amount_atoms_to_return,
+                    )?;
+                }
+                remove_order_from_tree_and_free(fixed, dynamic, current_order_index, !is_bid)?;
+                current_order_index = next_order_index;
+                continue;
+            }
+
+            // Stop trying to match if price no longer satisfies limit.
+            if (is_bid && other_order.get_price() > price)
+                || (!is_bid && other_order.get_price() < price)
+            {
+                break;
+            }
+
+            // Got a match. First make sure we are allowed to match.
+            assert_can_take(order_type)?;
+
+            if order_type == OrderType::PostOnlySlide {
+                // Post only slide creates a zero spread book.
+                maker_price = other_order.get_price();
+                break;
+            }
+
+            // Match the order
+            let other_trader_index: DataIndex = other_order.get_trader_index();
+            let did_fully_match_resting_order: bool =
+                remaining_base_atoms >= other_order.get_num_base_atoms();
+            let matched_num_base_atoms: BaseAtoms = if did_fully_match_resting_order {
+                other_order.get_num_base_atoms()
+            } else {
+                remaining_base_atoms
+            };
+
+            let matched_price: QuoteAtomsPerBaseAtom = other_order.get_price();
+
+            // Round in favor of the maker. There is a later check that this
```

Comment:
Clarify this is max 0.999.. atoms, correct?

Reply:
Yes. The rounding is always less than 1 atom